mne/preprocessing/annotate_amplitude.py

```diff
@@ -0,0 +1,227 @@
+    # group
+    annotations = annotation_flat + annotation_peak
+    # bads
+    bads = [raw.ch_names[bad] for bad in bads if bad not in raw.info['bads']]
+
+    return annotations, bads
+
+
+def _check_ptp(ptp, name, info, picks):
+    """Check the PTP threhsold argument, and converts it to dict if needed."""
+    _validate_type(ptp, ('numeric', dict, None))
+
+    if ptp is not None and not isinstance(ptp, dict):
+        if ptp < 0:
+            raise ValueError(
+                f"Argument '{name}' should define a positive threshold. "
+                f"Provided: '{ptp}'.")
+        ch_types = set(_get_channel_types(info, picks))
+        ptp = {ch_type: ptp for ch_type in ch_types}
+    elif isinstance(ptp, dict):
+        for key, value in ptp.items():
+            if value < 0:
+                raise ValueError(
+                    f"Argument '{name}' should define positive thresholds. "
+                    f"Provided for channel type '{key}': '{value}'.")
+    return ptp
+
+
+def _check_bad_percent(bad_percent):
+    """Check that bad_percent is a valid percentage and converts to float."""
+    _validate_type(bad_percent, 'numeric', 'bad_percent')
+    bad_percent = float(bad_percent)
+    if not 0 <= bad_percent <= 100:
+        raise ValueError(
+            "Argument 'bad_percent' should define a percentage between 0% "
+            f"and 100%. Provided: {bad_percent}%.")
+    return bad_percent
+
+
+def _check_min_duration(min_duration, raw_duration):
+    """Check that min_duration is a valid duration and converts to float."""
+    _validate_type(min_duration, 'numeric', 'min_duration')
+    min_duration = float(min_duration)
+    if min_duration < 0:
+        raise ValueError(
+            "Argument 'min_duration' should define a positive duration in "
+            f"seconds. Provided: '{min_duration}' seconds.")
+    if min_duration >= raw_duration:
+        raise ValueError(
+            "Argument 'min_duration' should define a positive duration in "
+            f"seconds shorter than the raw duration ({raw_duration} seconds). "
+            f"Provided: '{min_duration}' seconds.")
+    return min_duration
+
+
+def _reject_short_segments(arr, min_duration_samples):
+    """Check if flat or peak segments are longer than the minimum duration."""
+    assert arr.dtype == bool and arr.ndim == 2
+    for k, ch in enumerate(arr):
+        onsets, offsets = _mask_to_onsets_offsets(ch)
+        _mark_inner(arr[k], onsets, offsets, min_duration_samples)
+    return arr
+
+
+@jit()
+def _mark_inner(arr_k, onsets, offsets, min_duration_samples):
+    """Inner loop of _reject_short_segments()."""
+    for start, stop in zip(onsets, offsets):
+        if stop - start < min_duration_samples:
+            arr_k[start:stop] = False
+
+
+def _create_annotations(any_arr, type_, raw):
+    """Create the peak of flat annotations from the any_arr."""
+    assert type_ in ('peak', 'flat')
```

Comment:
this is fine, but FYI we often use `kind` (rather than `type_` or `_type`) in situations like this

Reply:
Ok, usually I like to use `_` after the variable name that I duplicate, I'll replace it with the standard you are used to seeing.